packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -411,6 +449,27 @@ export function makeVirtualObjectManager(
     }
   }
 
+  function copyMethods(behaviorTemplate) {
+    const obj = {};
+    for (const [name, func] of Object.entries(behaviorTemplate)) {
```

Comment:
This enables Symbol-named methods, yeah? I think we want that, at least in the long run, and anyways `Far` should be the judge of what keys are allowed, so I think `Object.entries` is exactly right.

Reply:
I wish. Two factors combine to make this not work well:
   * `Object.entries` omits non-enumerable properties
   * Object literals with symbol-named properties initialize those properties to be non-enumerable.

```js
> x = {[Symbol.asyncIterator]: () => null}
{ [Symbol(Symbol.asyncIterator)]: [Function: [Symbol.asyncIterator]] }
> Object.entries(x)
[]
> Object.getOwnPropertyDescriptors(x)
{
  [Symbol(Symbol.asyncIterator)]: {
    value: [Function: [Symbol.asyncIterator]],
    writable: true,
    enumerable: true,
    configurable: true
  }
}
```

Reply:
No, forget what I just said. The session I captured above shows that my second bullet is false. The new property is enumerable. But `Object.entries` omitted it anyway. Which corrected my dim memory: `Object.entries` only includes string-named properties.

In any case, I agree that `Far` should know what it accepts and others should make minimal assumptions about what that is. Use `Reflect.ownKeys` or `Object.getOwnPropertyDescriptors` in order to minimize such assumptions.

Reply:
So we are going to explicitly allow symbol-named methods here and leave it to `Far` to reject them?
What about symbol-named facets?

Reply:
Thinking about this further, symbol-named facets aren't straightforward, because I need to put them in order and sorting Symbols is a pain.  In any event, I don't think there's a use case for this aside from obsessive generality, so let's take the idea of symbol-named facets off the table for now.

Reply:
On deeper analysis, symbol-named methods are kind of PITA too, though if we think it's worth the effort I can do it.  But I really don't think it's worth the effort.

Reply:
Definitely no symbol-named facets. The record naming facets is a copyRecord, and the properties of a copyRecord must be string-named own enumerable properties, so they're exempt from all these painful differences between ways to enumerate own properties. So this restriction is architectural, not expedient.

> symbol-named methods here and leave it to `Far` to reject them?

`Far` does not reject symbol-named methods. Nor does it reject methods named by non-enumerable properties.

But we currently have only one case we need, `@@asyncIterator`, which we could continue to special case. If postponing general support for symbol-named methods gets us to MN1 faster, I'm all for postponing. But I think our distributed computational model should eventually support symbol-named methods in general.

Reply:
If I can put in symbol-named methods in less than, say, an hour's work I'll do it.  I suspect I probably can.

Reply:
A weird thought occurred to me as I was pondering implementation: as long as we do not support symbol-named methods in our over-the-wire message encoding, someone can use symbol-named methods as a kind of private-to-the-local-vat scoping technique.  I'm not saying this is a good idea, but I can imagine people doing it.  Then, when we turn that feature on, we break their imagined security perimeter.  (Even once we support symbols in the method selector portion of a message, I presume they would still be able to use non-registered, non-well-known symbols for this purpose.)  This makes me wonder if we should hold off implementing symbol-named methods until we are also implementing symbols as method selectors in messages.